drivers/sensor/grow_r502a/grow_r502a.c

```diff
@@ -0,0 +1,713 @@
+		LOG_ERR("Error receiving ack packet 0x%X", drv_data->buffer[6]);
+		return -EIO;
+	}
+
+	*template_cnt = drv_data->buffer[FPS_HEADER_LEN + 1];
+	*template_cnt <<= 8;
+	*template_cnt |= drv_data->buffer[FPS_HEADER_LEN + 2];
+
+	return drv_data->buffer[9];
+}
+
+static uint8_t fps_get_image(const struct device *dev)
+{
+	struct grow_r502a_data *drv_data = dev->data;
+	uint8_t data = FPS_GENIMAGE;
+
+	send_packet(dev, FPS_COMMAND_PACKET, &data, 1);
+
+	int ret = receive_data(dev, drv_data->buffer);
+
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (drv_data->buffer[6] != FPS_ACK_PACKET) {
+		LOG_ERR("Error receiving ack packet 0x%X", drv_data->buffer[6]);
+		return -EIO;
+	}
+
+	return drv_data->buffer[9];
+}
+
+static uint8_t fps_image2Tz(const struct device *dev, uint8_t slot)
+{
+	struct grow_r502a_data *drv_data = dev->data;
+	uint8_t data[2] = {FPS_IMAGE2TZ, slot};
+
+	send_packet(dev, FPS_COMMAND_PACKET, data, sizeof(data));
+
+	int ret = receive_data(dev, drv_data->buffer);
+
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (drv_data->buffer[6] != FPS_ACK_PACKET) {
+		LOG_ERR("Error receiving ack packet 0x%X", drv_data->buffer[6]);
+		return -EIO;
+	}
+
+	return drv_data->buffer[9];
+}
+
+static uint8_t fps_create_model(const struct device *dev)
+{
+	struct grow_r502a_data *drv_data = dev->data;
+	uint8_t data = FPS_REGMODEL;
+
+	send_packet(dev, FPS_COMMAND_PACKET, &data, 1);
+
+	int ret = receive_data(dev, drv_data->buffer);
+
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (drv_data->buffer[6] != FPS_ACK_PACKET) {
+		LOG_ERR("Error receiving ack packet 0x%X", drv_data->buffer[6]);
+		return -EIO;
+	}
+
+	return drv_data->buffer[9];
+}
+
+static uint8_t fps_store_model(const struct device *dev, uint16_t id)
+{
+	struct grow_r502a_data *drv_data = dev->data;
+	uint8_t data[4] = {FPS_STORE, 0x01, id >> 8, id & 0xFF};
+
+	send_packet(dev, FPS_COMMAND_PACKET, data, sizeof(data));
+
+	int ret = receive_data(dev, drv_data->buffer);
+
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (drv_data->buffer[6] != FPS_ACK_PACKET) {
+		LOG_ERR("Error receiving ack packet 0x%X", drv_data->buffer[6]);
+		return -EIO;
+	}
+
+	return drv_data->buffer[9];
+}
+
+static uint8_t fps_delete_model(const struct device *dev, uint16_t id)
+{
+	struct grow_r502a_data *drv_data = dev->data;
+	uint8_t data[5] = {FPS_DELETE, id >> 8, id & 0xFF, 0x00, 0x01};
+
+	send_packet(dev, FPS_COMMAND_PACKET, data, sizeof(data));
+
+	int ret = receive_data(dev, drv_data->buffer);
+
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (drv_data->buffer[6] != FPS_ACK_PACKET) {
+		LOG_ERR("Error receiving ack packet 0x%X", drv_data->buffer[6]);
+		return -EIO;
+	}
+
+	return drv_data->buffer[9];
+}
+
+static uint8_t fps_search(const struct device *dev, uint8_t slot,
+				uint16_t *finger_id, uint16_t *matching_score)
+{
+	struct grow_r502a_data *drv_data = dev->data;
+	uint8_t data[6] = {FPS_SEARCH, slot, 0x00, 0x00, (uint8_t)(FPS_DEFAULT_CAPACITY >> 8),
+				(uint8_t)(FPS_DEFAULT_CAPACITY & 0xFF)};
```

Comment:
another suggestion: seems that code can be formatted better. See https://github.com/zephyrproject-rtos/zephyr/pull/41307